pkg/model/trace/combine.go

```diff
@@ -58,12 +63,19 @@ func (c *Combiner) Consume(tr *tempopb.Trace) (int, error) {
 	return c.ConsumeWithFinal(tr, false)
 }
 
+// Consume the given partial trace, no further processing is needed
+func (c *Combiner) ConsumePartialTrace(tr *tempopb.Trace) (int, error) {
+	c.result = tr
```

Comment:
I guess we don't need to clone the struct here

Reply:
this is just copying a pointer and so won't clone the struct. regardless, i don't think this func should exist. we are very close. let's sync offline to push this through. we're not seeing eye to eye on some details we need to be aligned on.